File: 2132-convert-1d-array-into-2d-array/2132-convert-1d-array-into-2d-array.go
func construct2DArray(original []int, m int, n int) [][]int {
    if len(original) != m * n {
        return make([][]int, 0)
    }

    result := make([][]int, m)
    for i := range result {
        result[i] = make([]int, n)
    }

    currentIdx := 0

    for i := range m {
        for j := range n {
            result[i][j] = original[currentIdx]
            currentIdx++
        }
    }

    return result
}

File: 2132-convert-1d-array-into-2d-array/2132-convert-1d-array-into-2d-array.java
class Solution {
    public int[][] construct2DArray(int[] original, int m, int n) {
        if (m * n != original.length) return new int[0][0];

        int[][] result = new int[m][n];
        
        int currentIdx = 0;
        for (int row = 0; row < m; row++) {
            for (int col = 0; col < n; col++) {
                result[row][col] = original[currentIdx];
                currentIdx++;
            }
        }

        return result;
    }
}